src/Model/ModelTrait.php

```diff
@@ -193,20 +217,21 @@ private function getNewInstance(\ReflectionClass $class, mixed $value): mixed
 
     /**
      * @param mixed $value
+     * @param bool $excludeNulls
      * @return mixed
      */
-    private function getValueData(mixed $value): mixed
+    private function getValueData(mixed $value, bool $excludeNulls = false): mixed
```

Comment:
Private?

Reply:
This is one of those few cases where `private` is appropriate. This method is internal to clay and shouldn't be available to anything that uses the trait (models should be simple and avoid have business logic in them). Since I can't encapsulate it in quite that manner, the best I can do is mark it as internal (private) and prevent subclasses from having access